src/compile/data/dataflow.ts

```diff
@@ -91,44 +91,48 @@ export class DataFlowNode {
 }
 
 export class OutputNode extends DataFlowNode {
-
   private _source: string;
 
-  private _refcount = 0;
+  private _name: string;
 
   public clone(): this {
     const cloneObj = new (<any>this.constructor);
-    cloneObj._source = this._source;
     cloneObj.debugName = 'clone_' + this.debugName;
-    cloneObj._refcount = this._refcount;
+    cloneObj._source = this._source;
+    cloneObj.type = this.type;
+    cloneObj.refCounts = this.refCounts;
     return cloneObj;
   }
 
-  constructor(source: string, public readonly type: DataSourceType) {
+  constructor(source: string, public readonly type: DataSourceType, private readonly refCounts: Dict<number>) {
     super(source);
 
-    this._source = source;
+    this._source = this._name = source;
+
+    if (this.refCounts && !(this._name in this.refCounts)) {
```

Comment:
It is probably still possible due to a user's mistake?  (If they specify two sub-specs with the same name, possibly because they copy one to another?) 

We could handle this in a separate issue though.